libnavigation-ui/src/main/java/com/mapbox/navigation/ui/speed/SpeedLimitView.java

```diff
@@ -0,0 +1,214 @@
+    }
+    return border;
+  }
+
+  private LayerDrawable getSpeedDrawable(SpeedLimitSign speedLimitSign) {
+    LayerDrawable layerDrawable;
+    if (speedLimitSign == SpeedLimitSign.MUTCD) {
+      GradientDrawable[] layers = {
+          backgroundDrawable(GradientDrawable.RECTANGLE),
+          borderDrawable(GradientDrawable.RECTANGLE),
+          backgroundDrawable(GradientDrawable.RECTANGLE)
+      };
+      layerDrawable = new LayerDrawable(layers);
+      layerDrawable.setLayerInset(
+          0,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET
+      );
+      layerDrawable.setLayerInset(
+          1,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET
+      );
+      layerDrawable.setLayerInset(
+          2,
+          MUTCD_INNER_BACKGROUND_INSET,
+          MUTCD_INNER_BACKGROUND_INSET,
+          MUTCD_INNER_BACKGROUND_INSET,
+          MUTCD_INNER_BACKGROUND_INSET
+      );
+    } else {
+      GradientDrawable[] layers = {
+          backgroundDrawable(GradientDrawable.OVAL),
+          borderDrawable(GradientDrawable.OVAL),
+          backgroundDrawable(GradientDrawable.OVAL)
+      };
+      layerDrawable = new LayerDrawable(layers);
+      layerDrawable.setLayerInset(
+          0,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET,
+          VIENNA_MUTCD_OUTER_BACKGROUND_INSET
+      );
+      layerDrawable.setLayerInset(
+          1,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET,
+          VIENNA_MUTCD_BORDER_INSET
+      );
+      layerDrawable.setLayerInset(
+          2,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET,
+          VIENNA_INNER_BACKGROUND_INSET
+      );
+    }
+    return layerDrawable;
+  }
+
+  @SuppressLint("DefaultLocale")
+  private String getSpeedLimit(SpeedLimitSign sign, SpeedLimitUnit unit, int speedLimitKmph) {
+    if (sign == SpeedLimitSign.MUTCD) {
+      if (unit == SpeedLimitUnit.KILOMETRES_PER_HOUR) {
+        return getContext().getString(R.string.max_speed, speedLimitKmph);
+      } else {
+        double speed = Math.round(speedLimitKmph * KILO_MILES_FACTOR);
+        String formattedSpeed = String.format("%.0f", speed);
+        return getContext().getString(R.string.max_speed, Integer.parseInt(formattedSpeed));
+      }
+    } else {
+      if (unit == SpeedLimitUnit.KILOMETRES_PER_HOUR) {
+        return String.valueOf(speedLimitKmph);
+      } else {
+        double speed = Math.round(speedLimitKmph * KILO_MILES_FACTOR);
+        return String.format("%.0f", speed);
+      }
+    }
+  }
+
```

Comment:
NIT could we move these `public` methods up with the other ones? We normally organize the code in this particular order :point_right: `public`, `protected`, `private`, etc. trying to locate them in order of calling. It'd be great if we can restructure the code this way for easier reading.